### PR TITLE
Added rust slices

### PR DESCRIPTION
**Pull request by ATREAY**
### Description

This pull request includes details about Rust slices and fixes issue #3876.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:54]:
```suggestion
```
Don't use the Subheading 4 in the content.

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:72]:
```suggestion
```

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:48]:
```suggestion
    let my_array = [1, 2, 3, 4, 5];
```
Change this line or the name of the array from all codes as it contains plagiarism.

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:78]:
```suggestion
This Example demonstrates how to create a slice from index 2 to the end of the array.
```
Add this line before the code.

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:60]:
```suggestion
This example illustrates how to create a slice from the start of the array to index 3 (excluding the element at index 3).
```
Please Add this line before the code so that it describes what to expect from the code.

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:90]:
```suggestion
```

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:96]: Add this line before the code block.

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:102]: These lines contain plagiarism, please remove the plagiarism.

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:100]:
```suggestion
In Rust, slices can also be mutable, allowing for the modification of elements within the slice.
```

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:98]: Contains plagiarism, replace it with an appropriate subheading.

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:107]: Change the syntax as we can use Slices on other data structures too.

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:122]:
```suggestion
```

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:128]:
```suggestion
This example demonstrates how to modify an element in a mutable slice. In this case, the second element of the slice is changed from 3 to 10.
```
Add this line before the code block.

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:34]:
```suggestion
```
Don't use a subheading called Output as there's no need for it. But add a line before the output shell such as "The example will result in the following output:".

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Nice entry @ATREAY but change all the syntax throughout this entry as Slices can be used on other data structures too. Don't use Output as a subheading and some of this content contains plagiarism, please change those lines and try to avoid including plagiarism in the content from now on.

**Comment by ATREAY** [comment]
Hi @avdhoottt, I've implemented the changes you suggested. Could you please review it when you have a moment?

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:24]:
```suggestion
Where `data_structure` refers to the chosen data structure, `start_index` denotes the starting point, and `end_index` represents one past the last element of the slice. This syntax is applicable to diverse data structures, enabling the creation of slices from different collections.
```

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:36]:
```suggestion
The example will result in the following output:
```
Don't bold any sentence in the content.

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:56]:
```suggestion
The example will result in the following output:
```

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:62]:
```suggestion
This example illustrates how to create a slice from the start of the array to index 3 (excluding the element at index 3).
```
Add this line below this heading "Excluding the Start Index of a Slice".

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:80]: You should add the line describing the program before the code block. Add this line after this heading - "Excluding the End Index of a Slice".

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:74]:
```suggestion
The example will result in the following output:
```

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:94]:
```suggestion
The example will result in the following output:
```

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:122]:
```suggestion
The example will result in the following output:
```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@ATREAY Here are some final changes, please make them.

**Comment by ATREAY** [comment]
Hey @avdhoottt, I've updated the changes you suggested.

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:127]:
```suggestion
```

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:111]:
```suggestion
### Example

The following example demonstrates how to modify an element in a mutable slice. In this case, the second element of the slice is changed from 3 to 10.

```rust
```
The description of the example should be before the code block.

File: content/rust/concepts/slices/slices.md
Comment (avdhoottt) [inline content/rust/concepts/slices/slices.md:128]:
```suggestion
```

**Review by avdhoottt — APPROVED** [review]
PR is ready for the second review.

File: content/rust/concepts/slices/slices.md
Comment (dakshdeepHERE) [inline content/rust/concepts/slices/slices.md:14]:
```suggestion
**Slices** in Rust are references to a contiguous sequence of elements in memory. Referencing a portion of a collection without making a full copy of the data is possible through them. A view into arrays, vectors, or other sequences is provided by slices, enabling work with specific parts of the data.
```
We don't use pronouns in docs like you he they them etc..

File: content/rust/concepts/slices/slices.md
Comment (dakshdeepHERE) [inline content/rust/concepts/slices/slices.md:23]: Change this to bullet points like this:
- 1
- 2
- 3

File: content/rust/concepts/slices/slices.md
Comment (dakshdeepHERE) [inline content/rust/concepts/slices/slices.md:35]:
```suggestion
The above example will result in the following output:
```

File: content/rust/concepts/slices/slices.md
Comment (dakshdeepHERE) [inline content/rust/concepts/slices/slices.md:43]:
```suggestion
**Slices** in Rust offer a convenient method to omit start and end indexes, enabling the creation of slices without specifying both or either of these indexes.
```
same thing we don't use pronouns.

File: content/rust/concepts/slices/slices.md
Comment (dakshdeepHERE) [inline content/rust/concepts/slices/slices.md:57]:
```suggestion
The above example will result in the following output:
```

File: content/rust/concepts/slices/slices.md
Comment (dakshdeepHERE) [inline content/rust/concepts/slices/slices.md:47]:
```suggestion
This example illustrates how to create a slice from the start of the array to index `3` (excluding the element at index 3):
```

File: content/rust/concepts/slices/slices.md
Comment (dakshdeepHERE) [inline content/rust/concepts/slices/slices.md:65]:
```suggestion
This Example demonstrates how to create a slice from index `2` to the end of the array:
```

File: content/rust/concepts/slices/slices.md
Comment (dakshdeepHERE) [inline content/rust/concepts/slices/slices.md:75]:
```suggestion
The above example will result in the following output:
```

File: content/rust/concepts/slices/slices.md
Comment (dakshdeepHERE) [inline content/rust/concepts/slices/slices.md:93]:
```suggestion
The above example will result in the following output:
```

File: content/rust/concepts/slices/slices.md
Comment (dakshdeepHERE) [inline content/rust/concepts/slices/slices.md:123]:
```suggestion
The above example will result in the following output:
```

File: content/rust/concepts/slices/slices.md
Comment (dakshdeepHERE) [inline content/rust/concepts/slices/slices.md:111]:
```suggestion
The following example demonstrates how to modify an element in a mutable slice. In this case, the second element of the slice is changed from `3` to `10`:
```

File: content/rust/concepts/slices/slices.md
Comment (dakshdeepHERE) [inline content/rust/concepts/slices/slices.md:83]:
```suggestion
This example displays the complete slice of the array, including all elements:
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
I am requesting a few changes from your side. After this, we will move to merging. 🚀

**Comment by ATREAY** [comment]
Hey @dakshdeepHERE, I've updated the changes you suggested.

**Comment by dakshdeepHERE** [comment]
@ATREAY looks good to merge.

**Review by dakshdeepHERE — APPROVED** [review]

File: content/rust/concepts/slices/slices.md
Comment (HishamT) [inline content/rust/concepts/slices/slices.md:14]: All three sentences say the same thing. Consider rewording this part.

File: content/rust/concepts/slices/slices.md
Comment (HishamT) [inline content/rust/concepts/slices/slices.md:28]:
```suggestion
> **Note:** This syntax is applicable to many forms of data, enabling the creation of slices from different collections.

File: content/rust/concepts/slices/slices.md
Comment (HishamT) [inline content/rust/concepts/slices/slices.md:48]:
```suggestion
When creating a slice in Rust, one can include both the start and end index or one can include only one of the indexes.
```

File: content/rust/concepts/slices/slices.md
Comment (HishamT) [inline content/rust/concepts/slices/slices.md:106]:
```suggestion
In Rust, when a slice is designated as mutable, the alteration of the elements within the slice is allowed. The creation of a mutable slice is done by using the `&mut` keyword.
```

**Review by HishamT — CHANGES_REQUESTED** [review]
I have reviewed the entry and left some comments. 

Note: consider changing the names of the syntax and examples section to avoid the merge conflict.

**Comment by ATREAY** [comment]
Hey @HishamT, I've updated the changes you suggested.

**Comment by ATREAY** [comment]
Hi @dakshdeepHERE, just waiting on your approval—any more changes needed?

**Comment by dakshdeepHERE** [comment]
Hey @ATREAY, I will get back to you. We have to discuss the last part. Lint error is coming as ## Syntax is repeating. We have decide to shift the last part to separate entry or leave it here with different names.

**Comment by dakshdeepHERE** [comment]
@ATREAY Looks good to merge. Thanks for the entry. 🚀

**Pull request merged**
